File: Momento_2/Bloque_6/Person/C++/src/Staff.cpp
#include "../include/Staff.h"
#include <iostream>

Staff::Staff(const std::string &name, const std::string &adress, const std::string &school, double pay)
    : Person(name, adress), school(school), pay(pay) {}

std::string Staff::getSchool() const { 
    return school; 
}
double Staff::getPay() const { 
    return pay; 
}

void Staff::setSchool(const std::string &school) { 
    this->school = school; 
}

void Staff::setPay(double pay) { 
    this->pay = pay; 
}

std::ostream &operator<<(std::ostream &os, const Staff &staff) {
    os << "Staff[ " << Person(staff)
       << ", School: " << staff.getSchool()
       << ", Pay: " << staff.getPay() << " ]";
    return os;
}

File: Momento_2/Bloque_6/Person/JAVA/src/Staff.java
import java.time.LocalDate;

public class Staff extends Person {

    private String school;
    private double pay;

    public Staff(String name, String address, LocalDate fechaNacimiento, String school,
            double pay) {
        super(name, address, fechaNacimiento);
        this.school = school;
        this.pay = pay;
    }

    public String getSchool() {
        return school;
    }

    public double getPay() {
        return pay;
    }

    public void setSchool(String school) {
        this.school = school;
    }

    public void setPay(double pay) {
        this.pay = pay;
    }

    @Override
    public String toString() {
        return "Staff[" + super.toString() + ", school= '" + school + ", pay= " + pay + ']';
    }

}
